checkUtils: remove non-POSIX compliant features

mcpcpc opened the pull request:
### Description

Remove non-POSIX shell features from `checkUtils()` as these result in error during execution in a POSIX compliant shell (e.g. ash). 

### Changes

- [x] Remove `function`, `local `and `declare` as these are not supported/preferred features in a POSIX sh specification [1]. Furthermore, there is no suitable alternative solution (that i could find).
- [x] Change variables from read-only switch (-r) to explicit `readonly` command.
- [x] Change local variables to global variable type schema (e.g. all capitalized letters).
- [x] Change while-loop recursion from array elements to string elements as Arrays are not part of the POSIX specification [1]

### Sources

1. https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_09_05
2. https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html22

tomice commented:
I understand that because this is moving from a bash/ksh-style array to a POSIX-style loop, local scope has been omitted from this change as that is not a POSIX feature. However, `readonly` should be POSIX compliant (double-check this). As such, because everything is, or will be, moved to global scope, it might be prudent to change these now global vars to use `readonly`. It would also be prudent to start signaling that these are global. Usually, we reserve variables that are in caps as a cue that they are global i.e. `readonly MSG="foo"` instead of  `readonly msg="foo"`

tomice commented:
Other than the above, LGTM. The comments in my first post are more stylization than anything else.

mcpcpc commented:
> I understand that because this is moving from a bash/ksh-style array to a POSIX-style loop, local scope has been omitted from this change as that is not a POSIX feature. However, `readonly` should be POSIX compliant (double-check this). As such, because everything is, or will be, moved to global scope, it might be prudent to change these now global vars to use `readonly`. It would also be prudent to start signaling that these are global. Usually, we reserve variables that are in caps as a cue that they are global i.e. `readonly UTILS="foo"` instead of `readonly utils="foo"`

Makes sense.  Committed the global variable names and added the `readonly ` command to both variables (see updated "Changes" section in the initial post.  Also squashed the last few commits.
